### DISCUSSION TITLE
How to efficiently model hundreds of parallel 1D channels without junctions by THM module

### DISCUSSION THREAD

**Discussion by NorthMagic** [idaholab/moose#27941] [Q&A Modules: Thermal Hydraulics]
I was modeling a full core with hundreds of 1D flow channels by using THM. Firstly, I found THM needs a long time to initialize setting up, and the consuming time increases greatly with used number of threads.
![屏幕截图 2024-06-19 105802](https://github.com/idaholab/moose/assets/63328077/5c9e6836-32d1-4d86-ad05-312a4e7ce9a2)

Besides, the initialization time increases slightly with used number of processors.
<img width="862" alt="屏幕截图 2024-06-19 132735" src="https://github.com/idaholab/moose/assets/63328077/9858f1da-1c57-49cc-9460-a4f268fb6fcf">

Then, I tried to run this simulation with the command line of "mpiexec -n 4 ......". But I encountered an error with "EXIT CODE: 9" which was caused by out of memory. The computing machine is WSL with 32 GB RAM and one Intel Ultra 9 CPU. The number of DOFs of nonlinear system and auxiliary systems are 27,360 and 110,580 respectively. It seems small and should not exceed memory requirement.
<img width="359" alt="image" src="https://github.com/idaholab/moose/assets/63328077/5efe1957-7d7c-4c9b-8ca9-62fab5afb289">

When I used 1 or 2 processors with a single thread, the simulation was passed both with a consuming memory of about 5 GB. It seems conflict with the mentioned error by using 4 processors.
Is it possibile to effectively model hundreds of 1D flow channels by THM?

**Comment by NorthMagic** [comment]
This is the input file for test. I hope it could help you find out how the unexpected things happened.
[msre_thm.zip](https://github.com/user-attachments/files/15895957/msre_thm.zip)

**Comment by GiudGiud** [comment]
Hello

Please follow the profiling instructions to see what is taking a long time:
https://mooseframework.inl.gov/application_development/profiling.html

> Is it possibile to effectively model hundreds of 1D flow channels by THM?

yes I have seen it done.

Guillaume
